container-search/src/main/java/com/yahoo/search/dispatch/searchcluster/SearchCluster.java

```diff
@@ -303,7 +303,7 @@ public void pingIterationCompleted() {
             updateSufficientCoverage(group, sufficientCoverage);
             trackGroupCoverageChanges(i, group, sufficientCoverage, averageDocumentsInOtherGroups);
         }
-        if ( ! anyGroupsSufficientCoverage ) {
+        if ( ! anyGroupsSufficientCoverage && (sumOfActiveDocuments == 0)) {
             vipStatus.removeFromRotation(clusterId);
```

Comment:
I casn't follow this, why did you change this logic? It used to be quite clean and symmetric and now I'm confused.

Reply:
I have added some clarifying comments to the code. But in essence this will remove it from rotation when no groups have sufficient coverage and there are no documents to be served. In essence this is when all nodes in all groups are down.